cmd/main.go

```diff
@@ -125,6 +126,37 @@ func lint(l *thriftcheck.Linter, filenames []string) (thriftcheck.Messages, erro
 	return l.LintFiles(filenames)
 }
 
+func getFilenames(inputs []string) ([]string, error) {
+	var paths []string
+	for _, input := range inputs {
+		info, err := os.Stat(input)
+		if err != nil {
+			return nil, err
+		}
+
+		if info.IsDir() {
+			err = filepath.Walk(input, func(path string, info os.FileInfo, err error) error {
+				if err != nil {
+					return err
+				}
```

Comment:
Do you think we should return an error here and completely stop walking the directory tree, or should we ignore errors and only accumulate the files we're able to visit?

I can see benefits to each approach so I'm curious how you think about it.

Reply:
I am a "let it crash" believer. This input is not a parameter that we can control internally, but rather it is something that the user inputs and is very uncontrollable. So, rather than using various methods to help unpredictable input to fallback or ignore (because you can't avoid all sorts of corner cases and strange things that users might input), I am more inclined to ask the user to go back and correct their input.